From fcfdf0e2f8568e6f97bc7c67024d6d075cd45178 Mon Sep 17 00:00:00 2001
From: thrishalig <thrishalig@ami.com>
Date: Thu, 15 Sep 2022 15:21:52 +0530
Subject: [PATCH] In hwmon driver fan and pwm starts from 1 , as per spec it
 should start from 0, made changes as per specification and verified in sysfs
 path

Tested:

verfied in sysfs path

root@intel-obmc:/sys/class/hwmon/hwmon0# ls
device      fan2_input  fan5_input  fan8_input  of_node     pwm12       pwm15       pwm4        uevent
fan0_input  fan3_input  fan6_input  fan9_input  pwm0        pwm13       pwm2        pwm5
fan1_input  fan4_input  fan7_input  name        pwm1        pwm14       pwm3        subsystem

Signed-off-by: thrishalig <thrishalig@ami.com>
---
 drivers/hwmon/aspeed-g6-pwm-tacho.c | 68 ++++++++++++++---------------
 1 file changed, 34 insertions(+), 34 deletions(-)

diff --git a/drivers/hwmon/aspeed-g6-pwm-tacho.c b/drivers/hwmon/aspeed-g6-pwm-tacho.c
index f9bfc83b32fe..b7075f1ca28e 100644
--- a/drivers/hwmon/aspeed-g6-pwm-tacho.c
+++ b/drivers/hwmon/aspeed-g6-pwm-tacho.c
@@ -692,39 +692,40 @@ static umode_t fan_dev_is_visible(struct kobject *kobj,
 	return a->mode;
 }
 
-static SENSOR_DEVICE_ATTR(pwm1, 0644,
+static SENSOR_DEVICE_ATTR(pwm0, 0644,
 			show_pwm, set_pwm, 0);
-static SENSOR_DEVICE_ATTR(pwm2, 0644,
+static SENSOR_DEVICE_ATTR(pwm1, 0644,
 			show_pwm, set_pwm, 1);
-static SENSOR_DEVICE_ATTR(pwm3, 0644,
+static SENSOR_DEVICE_ATTR(pwm2, 0644,
 			show_pwm, set_pwm, 2);
-static SENSOR_DEVICE_ATTR(pwm4, 0644,
+static SENSOR_DEVICE_ATTR(pwm3, 0644,
 			show_pwm, set_pwm, 3);
-static SENSOR_DEVICE_ATTR(pwm5, 0644,
+static SENSOR_DEVICE_ATTR(pwm4, 0644,
 			show_pwm, set_pwm, 4);
-static SENSOR_DEVICE_ATTR(pwm6, 0644,
+static SENSOR_DEVICE_ATTR(pwm5, 0644,
 			show_pwm, set_pwm, 5);
-static SENSOR_DEVICE_ATTR(pwm7, 0644,
+static SENSOR_DEVICE_ATTR(pwm6, 0644,
 			show_pwm, set_pwm, 6);
-static SENSOR_DEVICE_ATTR(pwm8, 0644,
+static SENSOR_DEVICE_ATTR(pwm7, 0644,
 			show_pwm, set_pwm, 7);
-static SENSOR_DEVICE_ATTR(pwm9, 0644,
+static SENSOR_DEVICE_ATTR(pwm8, 0644,
 			show_pwm, set_pwm, 8);
-static SENSOR_DEVICE_ATTR(pwm10, 0644,
+static SENSOR_DEVICE_ATTR(pwm9, 0644,
 			show_pwm, set_pwm, 9);
-static SENSOR_DEVICE_ATTR(pwm11, 0644,
+static SENSOR_DEVICE_ATTR(pwm10, 0644,
 			show_pwm, set_pwm, 10);
-static SENSOR_DEVICE_ATTR(pwm12, 0644,
+static SENSOR_DEVICE_ATTR(pwm11, 0644,
 			show_pwm, set_pwm, 11);
-static SENSOR_DEVICE_ATTR(pwm13, 0644,
+static SENSOR_DEVICE_ATTR(pwm12, 0644,
 			show_pwm, set_pwm, 12);
-static SENSOR_DEVICE_ATTR(pwm14, 0644,
+static SENSOR_DEVICE_ATTR(pwm13, 0644,
 			show_pwm, set_pwm, 13);
-static SENSOR_DEVICE_ATTR(pwm15, 0644,
+static SENSOR_DEVICE_ATTR(pwm14, 0644,
 			show_pwm, set_pwm, 14);
-static SENSOR_DEVICE_ATTR(pwm16, 0644,
+static SENSOR_DEVICE_ATTR(pwm15, 0644,
 			show_pwm, set_pwm, 15);
 static struct attribute *pwm_dev_attrs[] = {
+	&sensor_dev_attr_pwm0.dev_attr.attr,
 	&sensor_dev_attr_pwm1.dev_attr.attr,
 	&sensor_dev_attr_pwm2.dev_attr.attr,
 	&sensor_dev_attr_pwm3.dev_attr.attr,
@@ -740,7 +741,6 @@ static struct attribute *pwm_dev_attrs[] = {
 	&sensor_dev_attr_pwm13.dev_attr.attr,
 	&sensor_dev_attr_pwm14.dev_attr.attr,
 	&sensor_dev_attr_pwm15.dev_attr.attr,
-	&sensor_dev_attr_pwm16.dev_attr.attr,
 	NULL,
 };
 
@@ -749,39 +749,40 @@ static const struct attribute_group pwm_dev_group = {
 	.is_visible = pwm_is_visible,
 };
 
-static SENSOR_DEVICE_ATTR(fan1_input, 0444,
+static SENSOR_DEVICE_ATTR(fan0_input, 0444,
 		show_rpm, NULL, 0);
-static SENSOR_DEVICE_ATTR(fan2_input, 0444,
+static SENSOR_DEVICE_ATTR(fan1_input, 0444,
 		show_rpm, NULL, 1);
-static SENSOR_DEVICE_ATTR(fan3_input, 0444,
+static SENSOR_DEVICE_ATTR(fan2_input, 0444,
 		show_rpm, NULL, 2);
-static SENSOR_DEVICE_ATTR(fan4_input, 0444,
+static SENSOR_DEVICE_ATTR(fan3_input, 0444,
 		show_rpm, NULL, 3);
-static SENSOR_DEVICE_ATTR(fan5_input, 0444,
+static SENSOR_DEVICE_ATTR(fan4_input, 0444,
 		show_rpm, NULL, 4);
-static SENSOR_DEVICE_ATTR(fan6_input, 0444,
+static SENSOR_DEVICE_ATTR(fan5_input, 0444,
 		show_rpm, NULL, 5);
-static SENSOR_DEVICE_ATTR(fan7_input, 0444,
+static SENSOR_DEVICE_ATTR(fan6_input, 0444,
 		show_rpm, NULL, 6);
-static SENSOR_DEVICE_ATTR(fan8_input, 0444,
+static SENSOR_DEVICE_ATTR(fan7_input, 0444,
 		show_rpm, NULL, 7);
-static SENSOR_DEVICE_ATTR(fan9_input, 0444,
+static SENSOR_DEVICE_ATTR(fan8_input, 0444,
 		show_rpm, NULL, 8);
-static SENSOR_DEVICE_ATTR(fan10_input, 0444,
+static SENSOR_DEVICE_ATTR(fan9_input, 0444,
 		show_rpm, NULL, 9);
-static SENSOR_DEVICE_ATTR(fan11_input, 0444,
+static SENSOR_DEVICE_ATTR(fan10_input, 0444,
 		show_rpm, NULL, 10);
-static SENSOR_DEVICE_ATTR(fan12_input, 0444,
+static SENSOR_DEVICE_ATTR(fan11_input, 0444,
 		show_rpm, NULL, 11);
-static SENSOR_DEVICE_ATTR(fan13_input, 0444,
+static SENSOR_DEVICE_ATTR(fan12_input, 0444,
 		show_rpm, NULL, 12);
-static SENSOR_DEVICE_ATTR(fan14_input, 0444,
+static SENSOR_DEVICE_ATTR(fan13_input, 0444,
 		show_rpm, NULL, 13);
-static SENSOR_DEVICE_ATTR(fan15_input, 0444,
+static SENSOR_DEVICE_ATTR(fan14_input, 0444,
 		show_rpm, NULL, 14);
-static SENSOR_DEVICE_ATTR(fan16_input, 0444,
+static SENSOR_DEVICE_ATTR(fan15_input, 0444,
 		show_rpm, NULL, 15);
 static struct attribute *fan_dev_attrs[] = {
+	&sensor_dev_attr_fan0_input.dev_attr.attr,
 	&sensor_dev_attr_fan1_input.dev_attr.attr,
 	&sensor_dev_attr_fan2_input.dev_attr.attr,
 	&sensor_dev_attr_fan3_input.dev_attr.attr,
@@ -797,7 +798,6 @@ static struct attribute *fan_dev_attrs[] = {
 	&sensor_dev_attr_fan13_input.dev_attr.attr,
 	&sensor_dev_attr_fan14_input.dev_attr.attr,
 	&sensor_dev_attr_fan15_input.dev_attr.attr,
-	&sensor_dev_attr_fan16_input.dev_attr.attr,
 	NULL
 };
 
-- 
2.25.1

